dbms/src/Functions/FunctionsDateTime.h

```diff
@@ -3277,6 +3277,42 @@ struct TiDBWeekOfYearTransformerImpl
     }
 };
 
+template <typename ToFieldType>
+struct TiDBToSecondsTransformerImpl
+{
+    static constexpr auto name = "tidbToSeconds";
+
+    static void execute(const Context & context,
+                        const ColumnVector<DataTypeMyTimeBase::FieldType>::Container & vec_from,
+                        typename ColumnVector<ToFieldType>::Container & vec_to,
+                        typename ColumnVector<UInt8>::Container & vec_null_map)
+    {
+        bool is_null = false;
+        for (size_t i = 0; i < vec_from.size(); ++i)
+        {
+            MyTimeBase val(vec_from[i]);
+            vec_to[i] = execute(context, val, is_null);
+            vec_null_map[i] = is_null;
+            is_null = false;
+        }
+    }
+
+    static ToFieldType execute(const Context & context, const MyTimeBase & val, bool & is_null)
+    {
+        // TiDB returns normal value if one of month/day is zero for to_seconds function, while MySQL return null if either of them is zero.
```

Comment:
Seems a TiDB bug, please fire a issue for this.

Reply:
OK，I'll fire one later.

Reply:
https://github.com/pingcap/tidb/issues/34629